### PR TITLE
StarkNet 0.11.0 - prep

### PR DESCRIPTION
**Pull request by JameStark**
### Description of the Changes

Creating a new structure for upcoming versions and version notes

### PR Preview URL

After you push a commit to this PR, a preview is built and a URL to the root of the preview appears in the comment feed.

Paste here the specific URL(s) of the content that this PR addresses.

### Check List

- [ ] Changes have been done against dev branch, and PR does not conflict
- [ ] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/182)

### CODE REVIEW

File: components/StarkNet/modules/starknet_versions/pages/version_notes.adoc [hunk 1]
```diff
@@ -5,10 +5,17 @@ The following version notes cover the ongoing version changes to StarkNet. To ge
 
 == StarkNet Environments
 
-Within StarkNet's deployment pipeline there are separate and distinct networks that operate independently of each other for testing before deployment. See https://docs.starknet.io/documentation/useful_info/[here] for more information. 
+Within StarkNet's deployment pipeline there are separate and distinct networks that operate independently of each other for testing before deployment. See https://docs.starknet.io/documentation/useful_info/[here] for more information.
```
Comment (stoobie) [inline components/StarkNet/modules/starknet_versions/pages/version_notes.adoc:8]: Rephrase this in terms of the user. Use second person.

Style guideline: Use this phrasing: For more information, see <topic_title>.

File: components/StarkNet/modules/starknet_versions/pages/version_notes.adoc [hunk 2]
```diff
@@ -5,10 +5,17 @@ The following version notes cover the ongoing version changes to StarkNet. To ge
...
 
+[id="version0.11.0"]
+== StarkNet v0.11.0 (Feb, 23)
 
-[id="versions"]
+[NOTE]
+====
+This is an upcoming version release.
```
Comment (stoobie) [inline components/StarkNet/modules/starknet_versions/pages/version_notes.adoc:15]: Treat `this` like an adjective. Include a noun after `this`.

**Review by stoobie — CHANGES_REQUESTED** [review]
A bunch of comments.

**Review by ArielElp — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/182)* status: 1 of 6 files reviewed, 11 unresolved discussions (waiting on @JameStark and @stoobie)

___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 2 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-UMBV5otq-hP1gstr:-NL-UMBV5otq-hP1gsts:bpviuv) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L2)):*
> ```AsciiDoc
> [id="upcoming_versions"]
> # StarkNet v0.11.0
> ```

I added in the notion a sentence under the json-rpc section and the mandatory message fees, let's add those here
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 72 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-S2SA9ZOVoJkcsWCK:-NL-S2SA9ZOVoJkcsWCL:b-yserq3) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L72)):*
> ```AsciiDoc
> After the class was declared, it can be used either in deploying new instances or via library calls. Note that to specify the class (either for deploy or library_call), one has to send `class_hash` and **NOT** `compiled_class_hash`. The sequencer and full nodes may internally maintain a mapping between the two, but a reference to a declared class has to be via its class hash.
> 
> ### Transaction hash
> ```

Let's write TBD here, it's not yet finalized



_Code quote:_
```AsciiDoc
### Transaction hash
```
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 103 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-UdLM-ttpciV-nrVK:-NL-UdLM-ttpciV-nrVL:buafgdd) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L103)):*
> ```AsciiDoc
> |flag|new nonce|n_updates
> 
> | 1 bit | 64 bits   | 64 bits   |
> ```

can we do this in Latex instead? You can take it from the data availability page and edit
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 111 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-UUVE6AoGk1Wl9K7p:-NL-UUVE6AoGk1Wl9K7q:b-oeuntg) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L111)):*
> ```AsciiDoc
> ## API Changes
> 
> ### Feeder Gateway
> ```

I added another change in the notion for the estimate_fee endpoint, let's add



_Code quote:_
```AsciiDoc
Feeder Gateway
```
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 116 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-U8Ib4PQsQaeszQ76:-NL-U8Ic0uKj1YXG3TrX:b-gfgib0) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L116)):*
> ```AsciiDoc
> - `**get_class_by_hash**` - the structure of the returned class will depend on whether or not it is a new type of class, compiled from Cairo 1.0.
> - `**get_full_contract**` - same as above
> - `**get_state_update`**
> ```

hierarchy needs to be fixed here for the items below (all part of get_state_update)
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 121 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-UD1X-zLXVc0nRmq8:-NL-UD1X-zLXVc0nRmq9:b-2guby3) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L121)):*
> ```AsciiDoc
> - `old_declared_contracts` is a list of class hashes
> - `declared_classes` is a dictionary that maps `class_hash` to `compiled_class_hash`
> - `**get_block**`
> ```

same here
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 124 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-UE1x59NgAi8NuSkC:-NL-UE1x59NgAi8NuSkD:b-2guby3) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L124)):*
> ```AsciiDoc
> - the `transactions` field is subject to the same changes in `get_transaction`
> - `state_root` is renamed to `state_commitment`
> - `**get_transaciton**`
> ```

same here
___
*[`components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc` line 130 at r1](https://reviewable.io/reviews/starknet-io/starknet-docs/182#-NL-UIvyE1U-k5b-zUjJ:-NL-UIvyE1U-k5b-zUjK:b-2guby3) ([raw file](https://github.com/starknet-io/starknet-docs/blob/ed427407658f0db24961a554842fca2bdba978c5/components/StarkNet/modules/starknet_versions/pages/upcoming_versions.adoc#L130)):*
> ```AsciiDoc
> ### Gateway
> 
> - `**add_transaction**`
> ```

same here

**Pull request closed without merging**